default/scripting/species/common/xenophobic.macros

```diff
@@ -6,46 +6,49 @@ XENOPHOBIC_SELF
                     Not Contains Building name = "BLD_CONC_CAMP"
-                    WithinStarlaneJumps jumps = 5 condition = Source
+                    Or [
+                        Not OwnedBy empire = RootCandidate.Owner
+                        Not EmpireHasAdoptedPolicy empire = Source.Owner name = "PLC_RACIAL_PURITY"
+                    ]
+                    WithinStarlaneJumps jumps = (NamedInteger name = "XENOPHOBIC_MAX_JUMPS" value = 4) condition = Source
                 ]
             ]
             stackinggroup = "XENOPHOBIC_LABEL_SELF"
             accountinglabel = "XENOPHOBIC_LABEL_SELF"
             priority = [[TARGET_AFTER_SCALING_PRIORITY]]
-            effects = [ 
-                SetTargetIndustry value = Value
-                    * (NamedReal name = "XENOPHOBIC_SELF_TARGET_INDUSTRY_SCALING" value = 0.9)
-                SetTargetResearch value = Value
-                    * (NamedReal name = "XENOPHOBIC_SELF_TARGET_RESEARCH_SCALING" value = 0.9)
-                SetTargetHappiness value = Value
-                    + (Statistic Count condition = And [
+            effects = SetTargetHappiness value = Value
+                + (NamedReal name = "XENOPHOBIC_TARGET_STABILITY_PERJUMP" value = -0.5)
+                * Statistic Sum
+                    value = (1 + NamedIntegerLookup name = "XENOPHOBIC_MAX_JUMPS"
+                        - JumpsBetween object = Target.ID object = LocalCandidate.ID)
+                    condition = And [
                         Planet
-                        OwnedBy empire = Source.Owner
-                        Not Or [ 
-                            Species name = Source.Species
-                            Species name = "SP_EXOBOT"
-                        ]
-                        Not Population high = 0
+                        VisibleToEmpire empire = Source.Owner
+                        Species
+                        Not Species name = Source.Species
+                        Not Species name = "SP_EXOBOT"
                         Not Contains Building name = "BLD_CONC_CAMP"
-                        WithinStarlaneJumps jumps = 5 condition = Source
-                    ]) * (NamedReal name = "XENOPHOBIC_SELF_TARGET_HAPPINESS_COUNT" value = -1.0)
-            ]
+                        Or [
+                            Not OwnedBy empire = RootCandidate.Owner
+                            Not EmpireHasAdoptedPolicy empire = Source.Owner name = "PLC_RACIAL_PURITY"
+                        ]
+                        WithinStarlaneJumps jumps = (NamedIntegerLookup name = "XENOPHOBIC_MAX_JUMPS") condition = Source
+                    ]
 
-        // Give a pop malus for self-sustaining xenophobic species based on the number of nearby planets with other species
+        // Pop malus for self-sustaining xenophobic species based on the number of nearby planets with other species
         EffectsGroup
             scope = Source
             activation = And [
```

Comment:
This hasn't changed, but since you're at it, please add
`(0 < [[XENOPHOBIC_SELFSUSTAINING_QUALIFYING_PLANET_COUNT]])`
to the activation to avoid showing a frenzy of 0.0.